### PR TITLE
Fix #1803: Support for raw JS types in use(x).as[T]

### LINKED ISSUE #1803: Extend use(x).as[T] to support js.Any source types

### PR DESCRIPTION
**Pull request by gzm0**
We reuse the infrastructure to extract method information from raw JS
types to typecheck structural conformance between two raw JS types.

### CODE REVIEW

**Comment by gzm0** [comment]
Review by @nicolasstucki
@sjrd, I'll probably hold this off until you have a look anyways (no rush though).

File: library/src/main/scala/scala/scalajs/macroimpls/UseAsMacros.scala [hunk 1]
```diff
@@ -67,43 +69,81 @@ private[scalajs] object UseAsMacros {
...
 
-          case JSMemberBracketAccess =>
-            c.abort(c.enclosingPosition, s"$trgTpe defines a " +
-                "@JSMemberBracketAccess method. Existence of such a method " +
-                "cannot be statically checked for any Scala exported type.")
+      for {
+        (jsMemberSelection, jsMembers) <- requiredMembers
+        jsMember <- jsMembers
+      } {
+
```
Comment (sjrd) [inline library/src/main/scala/scala/scalajs/macroimpls/UseAsMacros.scala]: Spurious blank line.

File: library/src/main/scala/scala/scalajs/macroimpls/UseAsMacros.scala [hunk 2]
```diff
@@ -67,43 +69,81 @@ private[scalajs] object UseAsMacros {
...
+          case _ =>
+        }
 
-          case JSMemberBracketCall =>
-            c.abort(c.enclosingPosition, s"$trgTpe defines a " +
-                "@JSMemberBracketCall method. Existence of such a method " +
-                "cannot be statically checked for any Scala exported type.")
+        val hasConformingMember = {
+          definedMembers.get(jsMemberSelection).fold(false) { candidates =>
+            candidates.exists(_.conformsTo(jsMember))
```
Comment (sjrd) [inline library/src/main/scala/scala/scalajs/macroimpls/UseAsMacros.scala]: I think `.getOrElse(..., Nil).exists(...)` would be clearer.

**Comment by sjrd** [comment]
Don't you have negative tests for this enhancement?

That's all.

File: library/src/main/scala/scala/scalajs/macroimpls/UseAsMacros.scala (hunk 2 above)
Comment (gzm0) [inline library/src/main/scala/scala/scalajs/macroimpls/UseAsMacros.scala]: Yes. Absolutely.

**Comment by gzm0** [comment]
Updated.

**Comment by gzm0** [comment]
Jenkins, test this please

**Comment by sjrd** [comment]
LGTM

**Comment by sjrd** [comment]
retest this please

**Pull request merged**